File: 20180711/Majority Element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # if len(nums) == 1:
        #     return nums[0]
        # dic = {}
        # for i in nums:
        #     if i in dic:
        #         dic[i] += 1
        #         if dic[i] > len(nums)/2 :
        #             return i
        #     else:
        #         dic[i] = 1
        count = 0
        candidate = None
        for i in nums:
            if count == 0:
                candidate = i
            if candidate == i:
                count += 1
            else:
                count -= 1
        return candidate
        
        

File: 20180711/Majority Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 0;
        int candidate = nums[0];
        for(int i = 0; i < nums.length; i ++){
            if(count == 0){
                candidate = nums[i];
            }
            if (candidate == nums[i]){
                count += 1;
            }
            else{
                count -= 1;
            }
        }
        return candidate;
    }
}
